rpc/core/tx.go

```diff
@@ -174,6 +174,45 @@ func proveTx(height int64, index uint32) (types.TxProof, error) {
 	return txProof, nil
 }
 
+// ProveShares creates an NMT proof for a set of shares to a set of rows.
+func ProveShares(
+	_ *rpctypes.Context,
+	height int64,
+	startShare uint64,
+	endShare uint64,
+) (types.SharesProof, error) {
+	var (
+		pSharesProof tmproto.SharesProof
+		sharesProof  types.SharesProof
+	)
+	env := GetEnvironment()
+	rawBlock, err := loadRawBlock(env.BlockStore, height)
+	if err != nil {
+		return sharesProof, err
+	}
+	res, err := env.ProxyAppQuery.QuerySync(abcitypes.RequestQuery{
+		Data: rawBlock,
+		Path: fmt.Sprintf(consts.ShareInclusionProofQueryPath, startShare, endShare),
+	})
+	if err != nil {
+		return sharesProof, err
+	}
+	if res.Value == nil && res.Log != "" {
+		// we can make the assumption that for custom queries, if the value is nil
+		// and some logs have been emitted, then an error happened.
+		return types.SharesProof{}, errors.New(res.Log)
+	}
```

Comment:
What happens if the value is nil and the logs are empty?

Reply:
Good point, I don't have a strong reasoning around this. My conclusion was based on running the query a lot of times and seeing the outcome. What I noticed was that if a custom query fails, the value is nil and the logs contain some message